src/integrationTest/kotlin/com/pubnub/api/integration/FilesIntegrationTest.kt

```diff
@@ -1,151 +1,75 @@
 package com.pubnub.api.integration
 
-import com.pubnub.api.CommonUtils.randomChannel
-import com.pubnub.api.PubNub
-import com.pubnub.api.callbacks.SubscribeCallback
-import com.pubnub.api.enums.PNStatusCategory
-import com.pubnub.api.models.consumer.PNStatus
-import com.pubnub.api.models.consumer.files.PNFileUploadResult
-import com.pubnub.api.models.consumer.pubsub.PNMessageResult
-import com.pubnub.api.models.consumer.pubsub.PNPresenceEventResult
-import com.pubnub.api.models.consumer.pubsub.PNSignalResult
-import com.pubnub.api.models.consumer.pubsub.files.PNFileEventResult
-import com.pubnub.api.models.consumer.pubsub.message_actions.PNMessageActionResult
-import com.pubnub.api.models.consumer.pubsub.objects.PNObjectEventResult
-import org.junit.Assert
-import org.junit.Test
-import java.io.ByteArrayInputStream
+import com.google.gson.JsonObject
+import com.pubnub.api.SpaceId
+import com.pubnub.api.models.consumer.MessageType
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.hasItem
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ValueSource
 import java.io.InputStream
-import java.nio.charset.StandardCharsets
-import java.util.Scanner
-import java.util.concurrent.CountDownLatch
+import java.nio.charset.Charset
 import java.util.concurrent.TimeUnit
-import java.util.concurrent.atomic.AtomicReference
+import org.hamcrest.Matchers.`is` as iz
 
 class FilesIntegrationTest : BaseIntegrationTest() {
-    @Test
-    fun uploadListDownloadDeleteWithCipher() {
-        uploadListDownloadDelete(true)
-    }
 
-    @Test
-    fun uploadListDownloadDeleteWithoutCipher() {
-        uploadListDownloadDelete(false)
-    }
+    @ParameterizedTest
+    @ValueSource(strings = ["enigma", ""])
+    fun sendListDownloadDelete(cipher: String) {
```

Comment:
I like code in this test. IMHO is sophisticated but still bit difficult for me to read. Maybe because of limited experience with Kotlin.